File: dart/lib/src/nft/get_domain_mint.dart
import 'package:solana/solana.dart' hide RpcClient;
import '../constants/addresses.dart';

/// Gets the mint address of a tokenized domain's NFT
///
/// This function mirrors js/src/nft/getDomainMint.ts
///
/// [domain] - The domain public key
///
/// Returns the NFT mint public key
Future<Ed25519HDPublicKey> getDomainMint(Ed25519HDPublicKey domain) async {
  const mintPrefix = 'tokenized_name';

  final seeds = [
    mintPrefix.codeUnits,
    domain.bytes,
  ];

  final programId = Ed25519HDPublicKey.fromBase58(nameTokenizerAddress);

  final result = await Ed25519HDPublicKey.findProgramAddress(
    seeds: seeds,
    programId: programId,
  );

  return result;
}

File: js/src/nft/getDomainMint.ts
import { PublicKey } from "@solana/web3.js";
import { MINT_PREFIX, NAME_TOKENIZER_ID } from "./const";

/**
 * Derives the NFT mint PDA for a tokenized SNS name account.
 *
 * @param domain Tokenized SNS name account address.
 * @returns The derived NFT mint address.
 *
 * @example
 * ```ts
 * const mint = getDomainMint(domainAddress);
 * ```
 */
export const getDomainMint = (domain: PublicKey) => {
  const [mint] = PublicKey.findProgramAddressSync(
    [MINT_PREFIX, domain.toBuffer()],
    NAME_TOKENIZER_ID,
  );
  return mint;
};
